service_container/3.3-di-changes.rst

```diff
@@ -468,6 +470,54 @@ Many autoconfigured tags have an optional priority. If you need to specify a pri
 (or any other optional tag attribute), no problem! Just :ref:`manually configure your service <services-manually-wire-args>`
 and add the tag. Your tag will take precedence over the one added by auto-configuration.
 
+5) Auto-configure with _instanceof
+----------------------------------
+
+And the final big change is ``_instanceof``. It acts as a default definition
+template (see `service-33-default_definition`_), but only for services whose
+class matches a defined one.
+This can be very useful when many services share some tag that cannot be
+inherited from an abstract definition:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            # ...
+
+            _instanceof:
+                class: AppBundle\Domain\LoaderInterface
+                public: true
+                tags: ['app.domain_loader']
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <instanceof id="AppBundle\Domain\LoaderInterface" public="true">
+                    <tag name="app.domain_loader" />
+                </instanceof>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use AppBundle\Domain\LoaderInterface;
+
+        $container->registerForAutoconfiguration(LoaderInterface::class)
+            ->addTag('app.domain_loader');
```

Comment:
And here I rewrote the whole thing, it looked like it was an unfinished copy/paste. I no longer use `setInstanceofConditionals`, not sure what's best, please tell me.

Reply:
It's ok to use this shortcut but unlike other formats it's missing `// ...` at the beginning to remind classes should be auto-registered first before being auto configured.

Maybe this function could be better documented with some extra comment, i.e: "This method returns a child definition to define the default configuration of the given class/interface.`.

Reply:
Thanks, addressed!